paymentsheet/src/test/java/com/stripe/android/utils/screenshots/PaparazziRule.kt

```diff
@@ -0,0 +1,169 @@
+            )
+
+            paparazzi.prepare(newDescription)
+
+            try {
+                val deviceConfig = testCase.apply(defaultDeviceConfig)
+                if (deviceConfig != defaultDeviceConfig) {
+                    paparazzi.unsafeUpdateConfig(deviceConfig)
+                }
+
+                paparazzi.snapshot {
+                    PaymentsTheme(colors = PaymentsTheme.getColors(testCase.isDarkTheme)) {
+                        Surface(color = MaterialTheme.colors.surface) {
+                            Box(
+                                contentAlignment = Alignment.Center,
+                                modifier = Modifier.padding(padding),
+                            ) {
+                                content()
+                            }
+                        }
+                    }
+                }
+            } finally {
+                paparazzi.close()
+            }
+        }
+    }
+
+    private fun createPaparazziDeviceConfig(): DeviceConfig {
+        return DeviceConfig.PIXEL_6.copy(
+            // Needed to shrink the screenshot to the height of the composable
+            screenHeight = 1,
+            softButtons = false,
+        )
+    }
+
+    private fun createPaparazzi(deviceConfig: DeviceConfig): Paparazzi {
+        return Paparazzi(
+            deviceConfig = deviceConfig,
+            // Needed to shrink the screenshot to the height of the composable
+            renderingMode = SessionParams.RenderingMode.V_SCROLL,
+            // Needed to make Paparazzi work in our API 33 project for now
+            environment = detectEnvironment().copy(
+                platformDir = "${androidHome()}/platforms/android-32",
+                compileSdkVersion = 32,
+            ),
+        )
+    }
+
+    companion object {
+        init {
+            makePaparazziWorkForApi33()
+        }
+    }
+}
+
+private fun makePaparazziWorkForApi33() {
+    // Temporary workaround to fix an issue with Paparazzi on API 33
+    // See: https://github.com/cashapp/paparazzi/issues/631#issuecomment-1326051546
+    val field = Build.VERSION::class.java.getField("CODENAME")
+    val newValue = "REL"
+
+    Field::class.java.getDeclaredField("modifiers").apply {
+        isAccessible = true
+        setInt(field, field.modifiers and ReflectionModifier.FINAL.inv())
+    }
+
+    field.apply {
+        isAccessible = true
+        set(null, newValue)
+    }
+}
+
+private fun Array<out Array<out PaparazziConfigOption>>.toTestCases(): List<TestCase> {
+    return createPermutations(this).map { TestCase(it) }
+}
+
+private fun createPermutations(
+    options: Array<out Array<out PaparazziConfigOption>>,
+): List<List<PaparazziConfigOption>> {
+    @Suppress("UNCHECKED_CAST")
+    return (options.toSet()).fold(listOf(listOf())) { acc, set ->
+        acc.flatMap { list -> set.map { element -> list + element } }
+    }
+}
+
+private data class TestCase(
```

Comment:
Nice work on this refactor. Makes things easier to understand in the snapshot function by leaps and bounds!